components/component.go

```diff
@@ -87,6 +89,30 @@ type ComponentInterface interface {
 	SetImageParamsMap(imageMap map[string]string) map[string]string
 	OverrideManifests(platform string) error
 	UpdatePrometheusConfig(cli client.Client, enable bool, component string) error
+	ConfigLogger(dscispec *dsciv1.DSCInitializationSpec) *zap.Logger
+}
+
+// by default, ConfigLogger set InfoLevel and Caller to InfoLevel, Stacktrace to ErrorLevel (close to what we have now)
+// config to use devel/development: change to DebugLevel, with Caller to DebugLevel, keep stacktrace to ErrorLevel
+// config to use prod/production: change to ErrorLevel, with Caller to ErrorLevel and keep stracetrace to ErrorLevel
+// not set a "default" in switch because support value in LogMode is enum.
+func (c *Component) ConfigLogger(dscispec *dsciv1.DSCInitializationSpec) *zap.Logger {
```

Comment:
Why the move to using the zap Logger directly? It was already used behind logr and logr is used everywhere else...?